### DISCUSSION TITLE
User Agent for Android App Not Working?

### DISCUSSION THREAD

**Discussion by Antkmk** [weather-gov/api#556] [Q&A]
I have an app, developed using Solar2d, that includes a weather feature. The iOS version of the app works, the Android version returns an "Access Denied" 403 error.
I've seen that a User-Agent is required. I wasn't explicitly providing one. I now am by sending ["User-Agent"] = "ADK46erNow/v2.6.0 (http://onebadant.com; ant@onebadant.com)"

That's the name and version of the app and the website and email address.

The URL I'm calling is:
"https://graphical.weather.gov/xml/SOAP_server/ndfdSOAPclientByDay.php?whichClient=NDFDgenByDay&lat=44.11&lon=-73.923725&format=24+hourly&startDate=2022-06-04&numDays=3&Unit=e&Submit=Submit"

I've been working with Solar2d on this issue, but the Solar2d forums and main developer haven't been able to tell me why this doesn't work. Can you see anything immediately wrong with what I'm doing? I can provide more detail if you need it.

Thank you for any assistance you can provide. ~Keith

**Comment by scadergit** [comment]
The 403 message should have a reference id. If you can provide I will look up the exact issue.

**Reply by Antkmk** [reply]
Here's what I get back from the call...(Reference #18.3c0ed317.1654542920.3c2758dc)

moto g(7): table: 0x79a9f941c0 {
                    moto g(7):  [responseHeaders] => table: 0x79a9f941c0
                    moto g(7):  {
                    moto g(7):     [Connection] => "close"
                    moto g(7):    
Jun 06 15:15:20.461  [Content-Length] => "331"
                    moto g(7):     [X-Android-Response-Source] => "NETWORK 403"
                    moto g(7):     [Date] => "Mon, 06 Jun 2022 19:15:20 GMT"
                    moto g(7):     [X-Android-Sent-Millis] => "1654542920117"
                    moto g(7):     [Expires] => "Mon, 06 Jun 2022 19:15:20 GMT"
                    moto g(7):     [X-Android-Selected-Protocol] => "http/1.1"
                    moto g(7):     [Strict-Transport-Security] => "max-age=31536000 ; includeSubDomains ; preload"
                    moto g(7):     [X-Android-Received-Millis] => "1654542920476"
                    moto g(7):     [C
Jun 06 15:15:20.461 ontent-Type] => "text/html"
                    moto g(7):     [Mime-Version] => "1.0"
                    moto g(7):     [HTTP-STATUS-LINE] => "HTTP/1.1 403 Forbidden"
                    moto g(7):     [Server] => "AkamaiGHost"
                    moto g(7):  }
                    moto g(7):  [responseType] => "text"
                    moto g(7):  [phase] => "ended"
                    moto g(7):  [bytesEstimated] => 331
                    moto g(7):  [response] => "<HTML><HEAD>
                    moto g(7): <TITLE>Access Denied</TITLE>
                    moto g(7): </HEAD><BODY>
                    moto g(7): <H1>Access Denied</H1>
                    moto g(7):  
                    moto g(7): You don't have permission to access "http&#58;&#47;&#47;graphical&#46;weather&#46;gov&#47;xml&#47;SOAP&#95;server&#47;ndfdSOAPclientByDay&#46;php&#63;" on this server.<P>
                    moto g(7): Reference&#32;&#35;18&#46;3c0ed317&#46;1654542920&#46;3c2758dc
                    moto g(7): </BODY>
                    moto g(7): </HTML>
                    moto g(7): "
                    moto g(7):  [name] => "networkRequest"
                    moto g(7):  [bytesTransferred] => 331
                    moto g(7):  [status] => 403
                    moto g(7):  [url] => "https://graphical.weather.gov/xml/SOAP_server/ndfdSOAPclientByDay.php?whichClient=NDFDgenByDay&lat=44.11&lon=-73.923725&format=24+hourly&startDate=2022-06-04&numDays=3&Unit=e&Submit=Submit"
                    moto g(7):  [isError] => false
                    moto g(7):  [requestId] => false
                    moto g(7): }

**Reply by scadergit** [reply]
At the time of that request, you weren't setting the custom user agent correctly, and the request was getting grouped into a larger rule that blocked the request.

This forum is for api.weather.gov, and not digital.weather.gov, which is [developed by MDL](https://vlab.noaa.gov/web/mdl). Glad I could help investigate this cause, but they would better serve your support concerns.

**Reply by Antkmk** [reply]
Thank you for your help. Is there any way you can show me what the user-agent of the request was? I know what I sent - listed above in my first message, but I'm wondering if Solar2d is changing or ignoring my setting. Thank you again! ~Keith
P.S. Also, I'm a little confused, so there's api.weather.gov, digital.weather.gov, and I'm calling graphical.weather.gov! :)

**Reply by scadergit** [reply]
It's reverting back to the default Android user agent.

**Reply by Antkmk** [reply]
Thank you! I'll pass this on to Solar2d. Sounds like they are ignoring my settings when I make the call! ~Keith
